src/jdk.httpserver/share/classes/sun/net/httpserver/UnmodifiableHeaders.java

```diff
@@ -26,14 +26,15 @@
 package sun.net.httpserver;
 
 import java.util.*;
+import java.util.function.BiFunction;
 import com.sun.net.httpserver.*;
 
-class UnmodifiableHeaders extends Headers {
+public class UnmodifiableHeaders extends Headers {
 
-        Headers map;
+        private Map<String, List<String>> map;
```

Comment:
How about:

```
private Headers headers;
private Map<String, List<String>> map;

public UnmodifiableHeaders(Headers map) {
    this.map = Collections.unmodifiableMap(map);
    this.headers = headers;
}
```

Then getFirst simply becomes `return headers.getFirst(key);`

Reply:
I guess this presumes that no reference to `headers` escapes? Can that be assumed?

Reply:
Yes - it's private - and that wasn't a problem before - was it?

Reply:
Added (back) a `headers` field as per you suggestion, which is only accessed by `getFirst`.